torch/_refs/__init__.py

```diff
@@ -2224,20 +2224,18 @@ def _fn(*args, out=None, **kwargs):
 @out_wrapper()
 def all(
     a: TensorLikeType,
-    dim: Optional[DimsType] = None,
+    dim: Optional[int] = None,
     keepdim: bool = False,
 ) -> TensorLikeType:
-    # Computes nelem
-    if isinstance(dim, Dim):
-        dim = (dim,)  # type: ignore[assignment]
-
-    a_ = _maybe_convert_to_dtype(a, torch.bool)
-    # avoid comparison with symbolic number of elements to make this op symint friendly
-    result = eq(sum(logical_not(a_), dim=dim, keepdim=keepdim), 0)
+    if dim is None:
+        result = torch.logical_not(torch.any(torch.logical_not(a)))
+    else:
+        result = torch.logical_not(
+            torch.any(torch.logical_not(a), dim, keepdim=keepdim)
+        )
```

Comment:
Can you leave a comment that this is written like this because all in core does not support `all(x, keepdim=True)`?

Reply:
Will do!

Reply:
I'm having some trouble updating this PR (using `ghstack`) so I may have to land this as-is unfortunately.

Reply:
oh, I had posted this comment before. Was not sure when I saw https://github.com/pytorch/pytorch/pull/110093 lol

ghstack should not cause issues (I haven't had any issue with it in more than a year). If you are hitting any issues ping internally any other PyTorch dev and they should be able to help you